### PR TITLE
Remove M.T/CT.Uwp/WinUI.UI dependency, internalize Test helpers

### LINKED ISSUE #39: Update Testing/Sample App Toolkit Dependencies
### Describe the bug

We kind of have a chicken-and-egg problem here. We needed some Toolkit code in order to make it nicer to work with our sample app and testing infrastructure while building it.

However, this code also lives in the Toolkit and needs to be ported/updated itself. This makes it hard to test and work with as we now have conflicts between the old version with dual namespaces and the new version with unified namespaces.

We need to solve two problems with this issue:
- [ ] Migrate to the initial unified namespace package (coming soon™)
- [ ] Provide a property flag `<ExcludeToolkitNuGetDependencies>true</ExcludeToolkitNuGetDependencies>` (or something) which guards including that package in the dependent tooling bits (tests/sample app) - would probably live in the common props in the root I guess? We'll have to think about where this needs to be set and how this works a bit more.

We need the later so that when we're actually building the sample app or tests within the single or all component projects which actually include the source for that code that we have a single reference. i.e. the source itself. Otherwise if we also include the package there'll be a conflict between the pre-built version and the source code (which we may have updated and want to test).

Therefore when the flag is enabled, we can put it in the `csproj` for any dependent packages we're using so that within the source repo, the source gets used, and for other things like `Labs` where that source doesn't live it'll pick up whatever last version of the package we've setup in the tooling module here.

### Steps to reproduce

```text
N/A
```

### Expected behavior

Able to work on/test code that we depend on in our tooling.

### Screenshots

_No response_

### Code Platform

- [ ] UWP
- [ ] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### Other Windows Build number

_No response_

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Other SDK version

_No response_

### Visual Studio Version

_No response_

### Visual Studio Build Number

_No response_

### Device form factor

_No response_

### Additional context

_No response_

### Help us help you

Yes, I'd like to be assigned to work on this item.

### PR DESCRIPTION
**Pull request by michael-hawker**
Fixes #39

Need to remove the dependency as that can't build the new version of it due to namespace conflicts with the global usings. Once we have a package version of the new unified version, we can investigate a better solution for building in both modes.

First step, will need a follow-on to fix this once we have a package.

Not sure if we just want to reference branch in other repo for now, or to merge into main.

Labs won't want to take this as it'll break a bunch of tests that use the Visual Tree helpers, so we may just want to wait until we have a solution for that?

### CODE REVIEW

**Comment by michael-hawker** [comment]
Ah, I forgot the example template uses the tree helper (and the compositiontargethelper) itself... hmmm

**Comment by michael-hawker** [comment]
Thinking we'll have this as a pin, and I'll re-work into the final PR once we have a package that we can unify around maybe?

**Comment by michael-hawker** [comment]
@Arlodotexe think this is ready for review? I can't think of what else we may need here at the moment? (outside of the CompositionTargetHelper, but we haven't brought those over yet.)

**Review by Arlodotexe — APPROVED** [review]
Needs docs, otherwise approved

File: .github/workflows/config/nuget.config [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<configuration>
+  <packageSources>
+    <clear />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+    <add key="MainLatest" value="https://pkgs.dev.azure.com/dotnet/CommunityToolkit/_packaging/CommunityToolkit-MainLatest/nuget/v3/index.json" protocolVersion="3" />
```
[the remark this answers is not in the document]
Comment (michael-hawker) [reply .github/workflows/config/nuget.config:5]: What do you suggest for documentation? Like calling out the `config` directory as the starting required files needed? I put that in the `tooling_readme.md` file here? Is it not sufficient?

**Pull request merged**